### ISSUE TITLE
SFTTrainer for non-packed dataset

### ISSUE THREAD

**Issue by talha1503** [huggingface/trl#1881]
Hi,
I am using the SFTTrainer to fine tune LLaMa 2 on this [dataset](https://huggingface.co/datasets/neil-code/dialogsum-test) . 
When I set `packing=False`  in the SFTTrainer, I get an error which says `[Unable to create tensor, you should probably activate truncation and/or padding with ‘padding=True’ ‘truncation=True’](https://discuss.huggingface.co/t/unable-to-create-tensor-you-should-probably-activate-truncation-and-or-padding-with-padding-true-truncation-true/98833)`.

I inspected the dataloader creation process in the SFTTrainer when we set `packing=True` and `packing=False` and noticed: 
1) When packing is set to True, the dataloader or the training dataset created by the trainer consists of only two columns: `input_ids `and `labels`. 
2) However, when packing is set to False, the training dataset created by the trainer consists of the columns from the huggingface dataset as well as two new fields: `input_ids` and `attention_mask` .

Even if `dataset_text_field` is provided when packing is set to False, the error still persists. I have tried using various combinations of formatting functions, still no difference. 

I have replicated the issue here on Google Colab: https://colab.research.google.com/drive/1gG6Y0x_YLnTjNETBavpmPhfs-QA3DsCw#scrollTo=IcSN3QWxtQCn 

Can someone help me out? If this issue persists, I am also interested in working on it, and come up with a PR.

**Comment by Galaxy-Husky** [comment]
Try set `remove_unused_columns` to True. I think the data collator can't do the padding for the string data of the huggingface dataset.

**Comment by qgallouedec** [comment]
Actually, the error comes from the fact that your dataset is not properly formatted. I'd recommend using the chat format instead (see examples)
